components/job-orchestration/job_orchestration/scheduler/compress/compression_scheduler.py

```diff
@@ -146,9 +155,31 @@ def _process_s3_input(
         paths_to_compress_buffer.add_file(object_metadata)
 
 
-def search_and_schedule_new_tasks(db_conn, db_cursor, clp_metadata_db_connection_config):
+def _fetch_existing_datasets(
+    db_cursor, clp_metadata_db_connection_config: Dict[str, Any]
+) -> Set[str]:
+    table_prefix = clp_metadata_db_connection_config["table_prefix"]
+    db_cursor.execute(f"SELECT name FROM `{table_prefix}{DATASETS_TABLE_SUFFIX}`")
+    rows = db_cursor.fetchall()
+    return {str(row["name"]) for row in rows}
+
+
+def search_and_schedule_new_tasks(
+    db_conn,
+    db_cursor,
+    clp_metadata_db_connection_config: Dict[str, Any],
+    clp_storage_engine: StorageEngine,
+    clp_archive_output: ArchiveOutput,
+    datasets_cache: Set[str],
+):
     """
     For all jobs with PENDING status, split the job into tasks and schedule them.
```

Comment:
Since we're modifying this function's docstring:
```suggestion
    For all jobs with PENDING status, splits the job into tasks and schedules them.
```